android/src/main/java/com/reactnative/pianoanalytics/PianoAnalyticsModule.kt

```diff
@@ -22,133 +23,111 @@ class RNPianoAnalyticsModule internal constructor(var context: ReactApplicationC
-//  old Method :    piano.deleteUser()
   }
 
   // PRIVACY MODE
   @ReactMethod
   override fun privacySetMode(mode: String) {
-    PianoAnalytics.getInstance().privacyModesStorage
-// DEPRECATED ?  utiliser consent a la place ?
-//    old Method : piano.privacySetMode(mode)
+    val privacyMode = when (mode) {
+      "exempt" -> PrivacyMode.EXEMPT
+      "optin" -> PrivacyMode.OPTIN
+      "optout" -> PrivacyMode.OPTOUT
+      else -> return
+    }
+    PianoAnalytics.getInstance().privacyModesStorage.currentMode = privacyMode
   }
 
-   @ReactMethod
-   override fun privacyGetMode(promise: Promise) {
-//   old Method :   piano.privacyGetMode { mode: String ->
-//       promise.resolve(mode)
-//     }
-   }
-
-   // VISITOR ID
-   @ReactMethod
-   override fun setVisitorId(visitorId: String) {
-     // est ce quil faut conserver cette method ?  la method get est visitorId // selon la doc  // The visitor identification is created the first time you use a pa.sendEvent() / pa.sendEvents() method.
-     PianoAnalytics.getInstance().customVisitorId = visitorId
-
-//  old Method :    piano.setVisitorId(visitorId)
-   }
-
-   @ReactMethod
-   override fun getVisitorId(promise: Promise) {
-     val visitorId = PianoAnalytics.getInstance().visitorId
+  @ReactMethod
+  override fun privacyGetMode(promise: Promise) {
+    val currentMode = PianoAnalytics.getInstance().privacyModesStorage.currentMode
+    promise.resolve(currentMode)
+  }
 
-     if (visitorId != null) {
-       promise.resolve(visitorId)
-     } else {
-       promise.reject("Fail to retrieve visitor id");
-     }
-   }
+  // VISITOR ID
+  @ReactMethod
+  override fun getVisitorId(promise: Promise) {
+    val visitorId = PianoAnalytics.getInstance().customVisitorId
+    if (visitorId != null) {
+      promise.resolve(visitorId)
+    } else {
+      promise.reject("Fail to retrieve custom visitor id")
+    }
+  }
 
-  // PRIVACY INCLUDE PROPERTY
   @ReactMethod
-  override fun privacyIncludeProperty(property: String, privacyModes: ReadableArray?, eventNames: ReadableArray?) {
-    var privacyModesList : List<String>? =  Arguments.toList(privacyModes)?.map { it.toString() }
-    var privacyModesArray: Array<String> = privacyModesList?.toTypedArray() ?: emptyArray()
+  override fun setVisitorId(visitorId: String) {
+    PianoAnalytics.getInstance().customVisitorId = visitorId
+  }
 
-    var eventNamesList: List<String>? = Arguments.toList(eventNames)?.map { it.toString() }
-    var eventNamesArray: Array<String> = eventNamesList?.toTypedArray() ?: emptyArray()
+  // PRIVACY INCLUDE PROPERTY
+  private fun includePropertiesInPrivacyModes(propertiesList: List<String>, privacyModesList: List<String>, eventNamesList: List<String>) {
+    for (privacyMode in privacyModesList) {
+      val mode = when (privacyMode) {
+        "exempt" -> PrivacyMode.EXEMPT
+        "optin" -> PrivacyMode.OPTIN
+        "optout" -> PrivacyMode.OPTOUT
+        else -> continue
```

Comment:
you can refactor this code in a function duplicated on privacySetMode methode

Reply:
cest fait